homeassistant/components/radio_browser/__init__.py

```diff
@@ -1,15 +1,76 @@
 """The Radio Browser integration."""
 from __future__ import annotations
 
-from radios import RadioBrowser, RadioBrowserError
+import mimetypes
 
+from radios import FilterBy, RadioBrowser, RadioBrowserError, Station
+import voluptuous as vol
+
+from homeassistant.components.media_source.error import Unresolvable
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import __version__
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_FAVORITE_RADIOS,
+    CONF_RADIO_BROWSER,
+    DOMAIN,
+    LAST_FAVORITE,
+    SERVICE_NEXT_RADIO,
+    SERVICE_PREV_RADIO,
+    SERVICE_START_RADIO,
+)
+from .media_source import CODEC_TO_MIMETYPE
+
+ATTR_PLAYER_ENTITY = "entity_id"
+ATTR_FAVORITE = "favorite"
+PLATFORMS = ["media_source"]
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_FAVORITE_RADIOS): cv.ensure_list,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
We don't allow new YAML configuration for integrations connecting to devices or services.

Reply:
Fair enough on the media browser point. 

What approach do you suggest as a way to easily specify a list of favorite stations? That is why I use the yaml config. I tried to implement it using config flow , but there is no support for lists of strings in that framework.

Reply:
You can make an architectural proposal of your change in the GitHub discussions in "architecture" repository (that is also in the Home Assistant organization).